src/test/java/org/opensearch/security/auditlog/compliance/RestApiComplianceAuditlogTest.java

```diff
@@ -41,21 +44,23 @@ public void testRestApiRolesEnabled() throws Exception {
             .build();
 
         setup(additionalSettings);
-        TestAuditlogImpl.clear();
-        String body = "{ \"password\":\"some new password\",\"backend_roles\":[\"role1\",\"role2\"] }";
-        HttpResponse response = rh.executePutRequest(
-            "_opendistro/_security/api/internalusers/compuser?pretty",
-            body,
-            encodeBasicHeader("admin", "admin")
-        );
-        Thread.sleep(1500);
-        Assert.assertEquals(HttpStatus.SC_CREATED, response.getStatusCode());
-        Assert.assertTrue(TestAuditlogImpl.messages.size() + "", TestAuditlogImpl.messages.size() == 1);
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("audit_request_effective_user"));
-        Assert.assertFalse(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_READ"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_WRITE"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("UPDATE"));
-        validateMsgs(TestAuditlogImpl.messages);
+        Thread.sleep(1000);
+        final List<AuditMessage> messages = TestAuditlogImpl.doThenWaitForMessages(() -> {
+            String body = "{ \"password\":\"some new password\",\"backend_roles\":[\"role1\",\"role2\"] }";
+            HttpResponse response = rh.executePutRequest(
+                "_opendistro/_security/api/internalusers/compuser?pretty",
+                body,
+                encodeBasicHeader("admin", "admin")
+            );
+            Assert.assertEquals(HttpStatus.SC_CREATED, response.getStatusCode());
+            try {
+                Thread.sleep(1500);
+            } catch (InterruptedException e) {
+                fail("Received unexpected exception");
+            }
+            Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("UPDATE"));
```

Comment:
The result of `doThenWaitForMessages(...)` is the list of audit messages that were generated for the requests, you can completely remove the sleeps and do all message validation using `messages` for assertions at or past line 62.

Reply:
Note; this applied to everywhere this pattern is used in these tests.

Reply:
Removed all sleeps